Legacy/bonej/src/main/java/org/bonej/plugins/ParticleDisplay.java

```diff
@@ -0,0 +1,477 @@
+			if (p > 0 && !surfacePoint.isEmpty()) {
+				Color3f pColour = new Color3f(0, 0, 0);
+				if (colourMode == GRADIENT) {
+					final float red = 1.0f - p / (float) nSurfaces;
+					final float green = 1.0f - red;
+					final float blue = p / (2.0f * nSurfaces);
+					pColour = new Color3f(red, green, blue);
+				} else if (colourMode == SPLIT) {
+					if (volumes[p] > splitValue) {
+						// red if over
+						pColour = new Color3f(1.0f, 0.0f, 0.0f);
+					} else {
+						// yellow if under
+						pColour = new Color3f(1.0f, 1.0f, 0.0f);
+					}
+				} else if (colourMode == ORIENTATION) {
+					pColour = ParticleDisplay.colourFromEigenVector(eigens[p]);
+				}
+				// Add the mesh
+				try {
+					univ.addTriangleMesh(surfacePoint, pColour, "Surface " + p).setLocked(true);
+				} catch (final NullPointerException npe) {
+					IJ.log("3D Viewer was closed before rendering completed.");
+					return;
+				}
+			}
+			p++;
+		}
+	}
+
+	static void display3DOriginal(final ImagePlus imp, final int resampling, final Image3DUniverse univ) {
+		final Color3f colour = new Color3f(1.0f, 1.0f, 1.0f);
+		final boolean[] channels = { true, true, true };
+		try {
+			univ.addVoltex(imp, colour, imp.getTitle(), 0, channels, resampling).setLocked(true);
+		} catch (final NullPointerException npe) {
+			IJ.log("3D Viewer was closed before rendering completed.");
+		}
+	}
+
+	// ----------------- HELPER METHODS --------------//
+
+	/**
+	 * Generate a colour based on the inertia tensor's eigenvector
+	 * 
+	 * Colour is from the HSB colour wheel scaled by 0.5 to fit into pi radians
+	 * (rather than the 2 pi it normally occupies), so that red is at 0, pi and 2pi
+	 * radians.
+	 * 
+	 * Colour is mapped to the axis-angle representation of the tensor so hue varies
+	 * as a function of second axis rotation around the first.
+	 * 
+	 * @param eigen Eigenvalue decomposition of the particle
+	 * @return Colour scaling in red for axis and green for angle
+	 */
+	static Color3f colourFromEigenVector(EigenvalueDecomposition eigen) {
+		final Matrix rotation = eigen.getV();
+
+		// deflection of long axis from image z axis, 0 - pi radians
+		final double angle = Math.acos(-Math.abs(rotation.get(2, 0)));
+
+		final float hue = (float) (angle / Math.PI);
+		final float saturation = 1.0f;
+		final float brightness = 1.0f;
+
+		final int rgb = Color.HSBtoRGB(hue, saturation, brightness);
+		final Color color = new Color(rgb);
+		float red = (float) (color.getRed() / 255d);
+		float green = (float) (color.getGreen() / 255d);
+		float blue = (float) (color.getBlue() / 255d);
+
+		return new Color3f(red, green, blue);
+	}
+
+	private static boolean isRadiiValid(final double[] radii) {
+		for (int r = 0; r < 3; r++) {
+			if (Double.isNaN(radii[r])) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	private static int clamp(int value, int min, int max) {
+		if (value < min)
+			return min;
+		if (value > max)
+			return max;
+		return value;
+	}
```

Comment:
`min` is always `0`, so
```
	private static int clamp(int value, int max) {
		if (value < 0)
			return 0;
		return Math.min(value, max);
	}
```

Reply:
You are correct for this particular use case. But as written it is more able to be upstreamed because it can handle any legal int. Optimising for a specific client method will make it brittle for another future client method. What it doesn't do is check to make sure that min < max.